Lib/doctest.py

```diff
@@ -1059,7 +1059,9 @@ def _get_test(self, obj, name, module, globs, source_lines):
         if module is None:
             filename = None
         else:
-            filename = getattr(module, '__file__', module.__name__)
+            # __file__ can be None for empty packages
+            filename = (getattr(module, '__file__', module.__name__) or
+                        module.__name__)
```

Comment:
```py
filename = getattr(module, '__file__', None) or module.__name__
```

Reply:
Thanks, this makes it simpler and seems to cover all the cases.

* `__file__` can have a value None for empty packages where it falls back to `module.__name__`. 
* `__file__` attribute may be absent where it uses None and then falls back to `module.__name__`
* `__file__` can have a value.